javaparser-core/src/main/java/com/github/javaparser/resolution/declarations/ResolvedReferenceTypeDeclaration.java

```diff
@@ -274,4 +274,21 @@ default boolean hasAnnotation(String qualifiedName) {
     }
 
     List<ResolvedConstructorDeclaration> getConstructors();
+
+
+    /**
+     * We don't make this _ex_plicit in the data representation because that would affect codegen
+     * and make everything generate like {@code <T extends Object>} instead of {@code <T>}
+     *
+     * @return true, if this represents {@code java.lang.Object}
+     * @see ResolvedReferenceType#isJavaLangObject()
+     * @see <a href="https://github.com/javaparser/javaparser/issues/2044">https://github.com/javaparser/javaparser/issues/2044</a>
+     */
+    default boolean isJavaLangObject() {
```

Comment:
this is used in several locations throughout the PR, and removes many of the duplicated and slightly differing attempts to compare against `java.lang.Object`